### DISCUSSION TITLE
manual mesh 3x3 VS ubl no probe 3x3

### DISCUSSION THREAD

**Discussion by Tapap84** [Jyers/Marlin#416] [Q&A] [answered]
Im still new to 3d printing and even newer to jyers firmware. I understand the manual mesh and I have my printer dialed in with it. I keep hearing people talk about using ubl. I dont have a probe or anything. Would I benefit from switching to the ubl from manual mesh?

**Comment by smooreace** [comment]
Manual tramming only gives you 4 points to adjust with and the nozzle
travels in a horizontal line, no matter how off your bed is... be it tilted
or warped like a sand dune. The gains from using a probe, either UBL or
BLTouch, is that it essentially creates a topographical map of the print
surface. The nozzle then follows that 'terrain'. The higher the frequency
of bed samples or measurements, the more accurate that terrain profile is.
So let's take an exaggerated example of a print bed that has a 1/2" dip in
the center.... you can tram the thing perfectly in the corners as you are
now. But printing anything in the center would lead to a spaghetti mess.
With UBL or BLT managed mesh the nozzle will follow that 1/2" dip and will
print perfectly (with a 1/2" bump on the bottom).

**Reply by Tapap84** [reply]
thanks for the info

**Comment by Jyers** [comment, marked as answer]
Read this: https://github.com/Jyers/Marlin/wiki/Configuration-Details

**Reply by Tapap84** [reply]
I see. It seems the manual mash version is going to do the best for me. Thanks

**Reply by Tapap84** [reply]
I have a question if you can. I just wanted to see the difference between the two. I ended up going back to the manual mesh version, thanks for adding the option to view the current saved values btw. I see that on the wiki it say the mesh for the UBL is saved in a different segment of the eeprom. When i go back to manual mesh version and make my mesh again, will there be a conflict of where it's pulling the save from or will it use the mesh created with manual mesh? Sorry if this sounds stupid, trying to educate myself more on firmware capabilities

**Reply by Jyers** [reply]
There shouldn't be any conflict, but it is always best practice to reset eeprom when flashing new firmware anyway.

**Reply by jetel73** [reply]
Hello, just to be sure, if my starting G-code is written that way ?

; Ender 3 Custom Start G-code
G92 E0 ; Reset Extruder
G29 A F2 ; Restore manual mesh level, Fade height 2mm
G28 ; Home all axes

Will the result be the same as my actual starting G-code below ?

; Ender 3 Custom Start G-code
G92 E0 ; Reset Extruder
M420 S1 Z2 ; Restore manual mesh level, Fade height 2mm
G28 ; Home all axes

Regards
jetel

**Reply by Jyers** [reply]
If you're using UBL, you'll also want to put a G29 L0 before your G29 A F2 in order to make sure you load the correct mesh

**Comment by smooreace** [comment]
Yes, except the second is using the existing stored mesh rather that
creating a new mesh before every print

**Reply by Jyers** [reply]
I believe he's using UBL, so G29 A F2 just enables leveling, it doesn't actually probe the bed like bilinear does on a G29

**Reply by jetel73** [reply]
You're right Jyers, I want to use the E3V2-UBL-NoProbe-5x5-v4.2.2-v1.3.4 driver, exactly like the Manual Mesh one. But with the possibility to : 
1. use the option to tilt my mesh when I need it, not on every print. 
2. Restore my mesh stored in the eeprom after flashing a new firmware

I'll probabely install it this week.

**Reply by Jyers** [reply]
Unfortunately manual UBL doesn't support mesh tilting. However it will allow you to restore your mesh after flashing, and store multiple meshes if need be!

**Reply by jetel73** [reply]
Restore the mesh is already a good thing... Thx
